testing/trino-benchto-benchmarks/src/test/java/io/trino/sql/planner/InMemoryMetastoreFactory.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    private void addStatistics(HiveMetastore metastore, SchemaTableName table, Optional<Column> partitionColumn)
+    {
+        if (partitionColumn.isPresent()) {
+            statisticsRepository.loadPartitionStatistics(table, partitionColumn.get().getName())
+                    .ifPresent(partitionStatistics -> {
+                        List<PartitionWithStatistics> partitions = partitionStatistics.entrySet().stream()
+                                .map(entry -> getPartitionWithStatistics(
+                                        table,
+                                        partitionColumn.get(),
+                                        entry.getKey(),
+                                        entry.getValue()))
+                                .collect(toImmutableList());
+                        metastore.addPartitions(table.getSchemaName(), table.getTableName(), partitions);
+                    });
+        }
+        else {
+            statisticsRepository.load(table).ifPresent(
+                    tableStatistics -> metastore.updateTableStatistics(
+                            table.getSchemaName(),
+                            table.getTableName(),
+                            NO_ACID_TRANSACTION,
+                            statistics -> tableStatistics));
+        }
+    }
+
+    private PartitionWithStatistics getPartitionWithStatistics(
+            SchemaTableName table,
+            Column partitionColumn,
+            String partitionName,
+            PartitionStatistics statistics)
+    {
+        Partition partition = Partition.builder()
+                .setDatabaseName(table.getSchemaName())
+                .setTableName(table.getTableName())
+                .setColumns(ImmutableList.of(partitionColumn))
+                .setValues(toPartitionValues(partitionName))
+                // Required by 'Partition', but not used by cost based planner tests.
+                .withStorage(storage -> storage.setStorageFormat(fromHiveStorageFormat(TEXTFILE)))
+                .build();
+        return new PartitionWithStatistics(partition, partitionName, statistics);
+    }
+
+    private Table getTable(SchemaTableName tableName, List<Column> columns, Optional<Column> partitionColumn)
+    {
+        return Table.builder()
+                .setDatabaseName(tableName.getSchemaName())
+                .setTableName(tableName.getTableName())
+                .setTableType(EXTERNAL_TABLE.name())
+                .setDataColumns(columns)
+                .setPartitionColumns(partitionColumn.map(ImmutableList::of).orElse(ImmutableList.of()))
+                // Required by 'Table', but not used by cost based planner tests.
+                .withStorage(storage -> storage
+                        .setStorageFormat(fromHiveStorageFormat(TEXTFILE))
+                        .setLocation(new Path(URI.create("file://" + createTempDirectory().getPath())).toString()))
```

Comment:
Does this location need to actually exist or any dummy path would work here ?

Reply:
This has to exist